src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/TokenCredentialOptions.cs

```diff
@@ -0,0 +1,123 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Azure.Core;
+using Azure.Identity;
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.Azure.WebJobs.Logging.ApplicationInsights
+{
+    public class TokenCredentialOptions
+    {
+        private const string AuthToken = "AAD";
+        private const string AuthAuthorizationKey = "Authorization";
+        private const string AuthClientIdKey = "ClientId";
+
+        public TokenCredentialOptions(string authenticationString)
+        {
+            if (!string.IsNullOrWhiteSpace(authenticationString))
+            {
+                var tokens = ParseAuthenticationString(authenticationString);
+
+                if (tokens.ContainsKey(AuthAuthorizationKey))
+                {
+                    Authorization = tokens[AuthAuthorizationKey];
+                }
+                if (tokens.ContainsKey(AuthClientIdKey))
+                {
+                    ClientId = tokens[AuthClientIdKey];
+                }
+            }
+        }
+        /// <summary>
+        /// The authentication kind.
+        /// </summary>
+        public string Authorization { get; set; }
+
+        /// <summary>
+        /// The client ID of an user-assigned identity.
+        /// </summary>
+        /// <remarks>
+        /// This must be specified if you're using user-assigned managed
+        /// identity.
+        /// </remarks>
+        public string ClientId { get; set; }
+
+        /// <summary>
+        /// Create an <see cref="TokenCredential"/> from an authentication string.
+        /// </summary>
+        /// <param name="value">
+        /// The authentication string containing semi-colon separated key=value tokens.</param>
+        /// <returns>New <see cref="ManagedIdentityCredential"/> from the authentication string.</returns>
+        public TokenCredential CreateTokenCredential()
+        {
+            switch (Authorization)
+            {
+                case AuthToken when string.IsNullOrWhiteSpace(ClientId):
+                    return new ManagedIdentityCredential();
+                case AuthToken when !string.IsNullOrWhiteSpace(ClientId):
+                    return new ManagedIdentityCredential(ClientId);
+                default:
+                    return null;
+            }
+        }
+
+        private IReadOnlyDictionary<string, string> ParseAuthenticationString(string value)
```

Comment:
Instead of returning a dictionary, can this method directly assign the properties as it parses? Also recommend this does some validation to ensure a good connection string was provided.

Reply:
Fixed.